src/components/AuthModal/index.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useAuthModal } from '../../context/AuthModalContext';
+import { OverlayWrapper } from '../OverlayWrapper';
+import { LoginForm } from './LoginForm';
+import { RegisterForm } from './RegisterForm';
+import { useAuth } from '../../context/AuthContext';
+
+export const AuthModal: React.FC = () => {
+  const { isUserLoggedIn } = useAuth();
+  const { closeAuthModal, isAuthModalOpen } = useAuthModal();
+
+  const [isRegisterForm, setIsRegisterForm] = useState(false);
+
+  const handleToggleForm = () => {
+    setIsRegisterForm(!isRegisterForm);
+  };
+
+  if (!isAuthModalOpen) {
+    return null;
+  }
+
+  if (isUserLoggedIn) {
+    closeAuthModal();
+  }
```

Comment:
Let's not make this a habit. Setting state in the render is fine for some edge cases when you wanna do a derivedStateFromProps, or you wanna stop the rendering before the commit phase. But those are STRAIGHT useState setter, not some random unknown-from-this-place function. 

This is none of those :D. Let's put it to useEffect. God forbid, some other developer, or you, in some time, will put some side effect into closeAuthModal - like tracking f.e. and it will spam that damn tracking on every render. Gosh!

Reply:
Ohhh, fixed and noted!! :O